### PR TITLE
Docker Compose Demo

### PR DESCRIPTION
**Pull request by cphalen**

### CODE REVIEW

File: content/lectures/lec03.md [hunk 1]
```diff
@@ -8,3 +8,60 @@ draft: false
...
+services:
+  docker_compose_demo_container:
+    image: docker_compose_demo_image:latest
+    command: node index.js
+    ports:
+      - "8080:8080"
+    volumes:
+      - type: bind
+        source: ./log
+        target: /etc/log
```
Comment (pawalt) [inline content/lectures/lec03.md:34]: putting this in `/etc` is a little sus. i'd just make a `/log` directory

**Review by pawalt — CHANGES_REQUESTED** [review]
Maybe add a version of the compose file with the `dockerfile` argument so builds happen automatically on `docker-compose up`

File: content/lectures/lec03.md (hunk 1 above)
Comment (cphalen) [reply content/lectures/lec03.md:34]: How about `/dev/null`? Sounds good thanks for pointing this out I'll do `/log`

File: content/lectures/lec03.md [hunk 2]
```diff
@@ -8,3 +8,54 @@ draft: false
...
+
+Note that Docker has a much easier time with this than a virtual machine would. When you share a volume between the host and a virtual machine, you have two separate kernels which both have to recognize it as a shared volume. Docker has an easier time because it's the same kernel on both processes, it just needs to expose the correct volume to the kernel.
+
+Feel free to play around with logging and other ways to interact with a shared volume. You can also change the configurations to share different volumes. You can also experiment with running two separate containers with a single Docker compose file, this is really when Docker compose starts to be a huge improvement over managing each container individually. This is something you'll get a lot of good experience with during homework two.
```
Comment (cphalen) [inline content/lectures/lec03.md:61]: I wrote homework two but actually meant homework one

**Pull request merged**